pandas/tests/test_complex.py

```diff
@@ -0,0 +1,122 @@
+import numpy as np
+import pytest
+
+import pandas as pd
+from pandas import (
+    Index,
+    Series,
+    DataFrame,
+)
+import pandas.core.algorithms as algos
+import pandas.util.testing as tm
+
+
+class TestComplexSupportBasic:
+    @pytest.mark.parametrize("array,expected", [
+        (
+            [1 + 1j, 0, 1, 1j, 1 + 2j],
+            Series([1, 1, 1, 1, 1], index=[1 + 2j, 1 + 1j, 1j, 1, 0])
+        ),
+        (
+            [1 + 2j, 0, 1j, 1, 1j, 1 + 1j],
+            # index is sorted by value counts in descending order by default
+            Series([2, 1, 1, 1, 1], index=[1j, 1 + 2j, 1 + 1j, 1, 0])
+        )
+    ])
+    def test_value_counts(self, array, expected):
+        result = algos.value_counts(array)
+        tm.assert_series_equal(result, expected)
+
+    @pytest.mark.parametrize("array,expected", [
+        (
+            [1 + 1j, 0, 1, 1j, 1 + 2j, 1 + 2j],
+            np.array([(1 + 1j), 0j, (1 + 0j), 1j, (1 + 2j)], dtype=object)
+        ),
+    ])
+    def test_unique(self, array, expected):
+        result = algos.unique(array)
+        tm.assert_numpy_array_equal(result, expected)
+
+    @pytest.mark.parametrize("array,expected", [
+        (
+            [0, 1j, 1j, 1, 1 + 1j, 1 + 2j, 1 + 1j],
+            Series([False, False, True, False, False, False, True], dtype=bool)
+        ),
+    ])
+    def test_duplicated(self, array, expected):
+        result = Series(array).duplicated()
+        tm.assert_series_equal(result, expected)
+
+    @pytest.mark.parametrize("array,expected", [
+        (
+            [0, 1j, 1j, 1, 1 + 1j, 1 + 2j, 1 + 1j],
+            Series([False, True, True, False, True, True, True], dtype=bool)
+        ),
+    ])
+    def test_isin(self, array, expected):
+        result = Series(array).isin([1j, 1 + 1j, 1 + 2j])
+        tm.assert_series_equal(result, expected)
+
+    @pytest.mark.parametrize("array,expected", [
+        (
+            [1, 2, 2 + 1j],
+            (np.array([0, 1, 2], dtype=np.int64),
+             np.array([(1 + 0j), (2 + 0j), (2 + 1j)], dtype=object))
+        ),
+    ])
+    def test_factorize(self, array, expected):
+        result = pd.factorize(array)
+        assert len(result) == 2
```

Comment:
I think easier to read at a quick glance if you just compare the two elements rather than the enumerating loop